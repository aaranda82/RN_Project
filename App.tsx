import axios from 'axios';
import React, {useState} from 'react';
import {
  Button,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';

import {Colors, Header} from 'react-native/Libraries/NewAppScreen';
import Section from './src/client/components/Section';

const App = () => {
  const [num, setNum] = useState(0);
  const [text, setText] = useState<Array<string>>([]);
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const handlePress = async () => {
    const res = await axios.get('http://localhost:5000');
    setText(prev => [...prev, res.data]);
    setNum(prev => prev + 1);
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <Header />
        <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
          }}>
          <Section title="Wow">
            <Text style={styles.highlight}>Great job!!</Text> You created a
            React-Native project.
          </Section>
          <Section title="Keep going">Now do more!</Section>
          <View>
            <Text style={styles.topText}>You have pressed the button</Text>
            <Text style={styles.text}>{num}</Text>
            <Text style={styles.text}>times</Text>
            <Button title="Press here" onPress={handlePress} />
            {text.map((t, i) => (
              <Text style={styles.text} key={t + i}>
                {t}
              </Text>
            ))}
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
  text: {textAlign: 'center'},
  topText: {textAlign: 'center', margin: 30},
});

export default App;
