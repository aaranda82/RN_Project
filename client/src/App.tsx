import React from 'react';
import { NavigationContainer } from '@react-navigation/native';

import Home from './components/Home/Home';
import OverView from './components/Overview/Overview';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import LoginForm from './components/LoginForm/LoginForm';
import RegisterForm from './components/LoginForm/RegisterForm';
import { RootStackParamList } from './Types';
import { storeModel } from './store';
import { StoreProvider } from 'easy-peasy';

const App = () => {
  const Stack = createNativeStackNavigator<RootStackParamList>();
  return (
    <NavigationContainer>
      <StoreProvider store={storeModel}>
        <Stack.Navigator initialRouteName="Home">
          <Stack.Screen name="Home" component={Home} />
          <Stack.Screen name="Login" component={LoginForm} />
          <Stack.Screen name="Register" component={RegisterForm} />
          <Stack.Screen
            name="Overview"
            component={OverView}
            options={{ headerLeft: undefined }}
          />
        </Stack.Navigator>
      </StoreProvider>
    </NavigationContainer>
  );
};

export default App;
