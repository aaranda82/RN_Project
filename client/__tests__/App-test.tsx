/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../src/App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.useFakeTimers();

it('renders app correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
