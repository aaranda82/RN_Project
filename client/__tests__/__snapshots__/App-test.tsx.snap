// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTScrollView
          contentInsetAdjustmentBehavior="automatic"
          style={
            Object {
              "backgroundColor": "#F3F3F3",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 24,
                      "fontWeight": "600",
                    },
                    Object {
                      "margin": 30,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Super Sweet Mobile App
              </Text>
              <Text
                style={
                  Object {
                    "color": "#566573",
                    "fontSize": 16,
                    "textAlign": "center",
                  }
                }
              >
                Create an account and log in
              </Text>
              <Text
                style={
                  Object {
                    "color": "#566573",
                    "fontSize": 16,
                    "textAlign": "center",
                  }
                }
              >
                to view all the 
                <Text
                  style={
                    Object {
                      "fontWeight": "600",
                    }
                  }
                >
                  super sweet
                </Text>
                 
                content
              </Text>
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Login
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Sign up
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
